### ISSUE TITLE
Mails from blacklisted mailserves not visible in quarantine and rspamd history

### ISSUE THREAD

**Issue by friedPotat0** [mailcow/mailcow-dockerized#2948]
**Prior to placing the issue, please check following:** *(fill out each checkbox with a `X` once done)*
- [X] I understand that not following below instructions might result in immediate closing and deletion of my issue.
- [X] I have understood that answers are voluntary and community-driven, and not commercial support.
- [X] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

---

**Description of the bug**: What kind of issue have you *exactly* come across?

Mails that are blocked using a public blacklist for mailservers (e.g. hostkarma.junkemailfilter.com) are not added to the quarantine. They are not even visible in the rspamd history.

At the moment a t-online mailserver (mailout08.t-online.de) is listed on multiple public blacklists (see https://mxtoolbox.com/SuperTool.aspx?action=blacklist%3amailout08.t-online.de&run=toolpage). Any mail that is sent from this mailserver is directly blocked by the mailcow mail system and the following response is transmitted from the t-online mailserver back to the sender:

> This is the mail system at host mailout08.t-online.de.
> 
> I'm sorry to have to inform you that your message could not
> be delivered to one or more recipients. It's attached below.
> 
> For further assistance, please send mail to postmaster.
> 
> If you do so, please include this problem report. You can
> delete your own text from the attached returned message.
> 
> The mail system
> 
> <xyz@example.com>: host
> mail.example.com[xxx.xxx.xxx.xxx] said: 550 5.7.1 Service
> unavailable; client [194.25.134.20] blocked using
> hostkarma.junkemailfilter.com (in reply to RCPT TO command)
> 
> [...]

**Reproduction of said bug**: How *exactly* do you reproduce the bug?

1. Remote mailserver is listed on a public blacklist
2. A sender sends a mail through the blacklisted mailserver to the mailcow mail system

__I have tried or I do...__ *(fill out each checkbox with a `X` if applicable)*
- [X] In case of WebUI issue, I have tried clearing the browser cache and the issue persists.
- [ ] I do run mailcow on a Synology, QNAP or any other sort of NAS.

**System information**

Further information (where applicable):

I will provide more information if neccessary to solve the problem.

| Question | Answer |
| --- | --- |
| My operating system | I_DO_REPLY_HERE |
| Is Apparmor, SELinux or similar active? | I_DO_REPLY_HERE |
| Virtualization technlogy (KVM, VMware, Xen, etc) | I_DO_REPLY_HERE |
| Server/VM specifications (Memory, CPU Cores) | I_DO_REPLY_HERE |
| Docker Version (`docker version`) | I_DO_REPLY_HERE |
| Docker-Compose Version (`docker-compose version`) | I_DO_REPLY_HERE |
| Reverse proxy (custom solution) | I_DO_REPLY_HERE |

Further notes:
 - Output of `git diff origin/master`, any other changes to the code? If so, please post them.
 - All third-party firewalls and custom iptables rules are unsupported. Please check the Docker docs about how to use Docker with your own ruleset. Nevertheless, iptabels output can help _us_ to help _you_: `iptables -L -vn`, `ip6tables -L -vn`, `iptables -L -vn -t nat` and `ip6tables -L -vn -t nat `
 - Check `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @172.22.1.254` (set the IP accordingly, if you changed the internal mailcow network) and `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @1.1.1.1` - output? Timeout?
 
 General logs:
- Please take a look at the [official documentation](https://mailcow.github.io/mailcow-dockerized-docs/debug-logs/).

**Comment by andryyy** [comment]
They are filtered by Postscreen and therefore before they ever reach a filter. :) You could disable those blacklists in Postscreen and just use Rspamd for filtering. But the RBLs are not complete in Rspamd (or are they?).

**Comment by friedPotat0** [comment]
Ok, then I will have a look at postscreen. Maybe it's enough to add the specific t-online mailserver to the whitelist, because it's been blacklisted for the second time already and I don't want to disable multiple blacklists completely.

Thanks for the quick reply!

**Comment by mkuron** [comment]
> the RBLs are not complete in Rspamd (or are they?)

We actually wouldn't want to drop postscreen in favor of dealing with RBLs in rspamd, @andryyy. rspamd requires significantly (orders of magnitude) more compute resources and memory per message than a pre-DATA rejection by postscreen, so depending on how much spam they get, many people would see a significant increase in server resource consumption and/or need to move to a more powerful server. As we discussed in another issue which I can't find right now, I don't think that would be a worthwhile tradeoff just to get quarantine.

I actually think T-Online should be automatically whitelisted by postscreen via https://github.com/mailcow/mailcow-dockerized/blob/master/data/conf/postfix/postscreen_access.cidr. Maybe that file is outdated (hasn't been modified in three years) or the postwhite project (https://github.com/stevejenkins/postwhite) does not consider T-Online trusted.

**Comment by friedPotat0** [comment]
I've came across the same thing during my research for a proper solution to my problem. The postscreen whitelist should definitely be updated.

The main problem with t-online is, that they don't use SPF in order to get listed by postwhite which crawls for SPF records. So for now I've manually added the t-online mailserver to the postscreen_access.cidr.

**Comment by andryyy** [comment]
@mkuron I know, I was just thinking about _him_ disabling it in Postfix. :) I agree it is probably best to keep it in Postscreen or at least wait for Rspamd to move the process of RBLs to a pre-filter stage.
